hack/chains/release-pipeline-with-ec-demo.sh

```diff
@@ -0,0 +1,117 @@
+#      link it to the pipeline ServiceAccount (so skopeo can copy the image there):
+#      oc create secret docker-registry release-demo quay.io --docker-username=$USER --docker-password=top-secret
+#      oc secrets link pipeline release-demo --for=pull,mount
+#
+#   2. Create a Secret named "cosign-public-key" with the "cosign.pub" attribute to house the
+#      public key to be used for verifying the signature of the image and its attestation.
+#
+#
+# Usage:
+#   release-pipeline-with-ec-demo.sh <src-image-ref> <dst-image-ref>
+# Where:
+#   <src-image-ref> string pointing to the image to be released, e.g.
+#                   quay.io/spam/bacon:latest
+#   <dst-image-ref> string pointing to the location where the image
+#                   image will be released to, e.g.
+#                   quay.io/spam/bacon:yummy
+# Environment Variables:
+#   TASK_BUNDLE image reference to the tekton bundle containing the
+#               verify-enterprise-contract task
+set -euo pipefail
+
+SRC_IMAGE_REF="$1"
+DST_IMAGE_REF="$2"
+
+NAMESPACE="$(oc get sa default -o jsonpath='{.metadata.namespace}')"
+SIG_KEY="k8s://$NAMESPACE/cosign-public-key"
+
+# The image used by the verify-enterprise-contract task. This image is usually
+# built by the 'hack/build-and-push.sh' script from the
+# https://github.com/redhat-appstudio/build-definitions repository.
+DEFAULT_TASK_BUNDLE='quay.io/lucarval/appstudio-tasks:63489f81a7680c2501b1c7e0802d24c6169d434e-2'
+TASK_BUNDLE="${TASK_BUNDLE:-${DEFAULT_TASK_BUNDLE}}"
+
+source $(dirname $0)/_helpers.sh
+
+
+title "Simple Release Pipeline"
+
+echo -n "
+---
+apiVersion: tekton.dev/v1beta1
+kind: Pipeline
+metadata:
+  name: simple-release
+spec:
+  params:
+  - name: SRC_IMAGE_REF
+    type: string
+    description: Image reference to verify
+  - name: DST_IMAGE_REF
+    type: string
+    description: Reference to copy the image to
+  - name: PUBLIC_KEY
+    type: string
+    description: >-
+      Public key used to verify signatures. Must be a valid k8s cosign
+      reference, e.g. k8s://my-space/my-secret where my-secret contains
+      the expected cosign.pub attribute.
+  tasks:
+  - name: ec
+    taskRef:
+      name: verify-enterprise-contract
+      bundle: ${TASK_BUNDLE}
+    params:
+    - name:  IMAGE_REF
+      value: \$(params.SRC_IMAGE_REF)
+    - name: PUBLIC_KEY
+      value: \$(params.PUBLIC_KEY)
+
+  - name: release
+    taskRef:
+      name: skopeo-copy
+      kind: ClusterTask
+    params:
+    - name: srcImageURL
+      value: docker://\$(params.SRC_IMAGE_REF)
+    - name: destImageURL
+      value: docker://\$(params.DST_IMAGE_REF)
+    runAfter:
+    - ec
+    workspaces:
+    - name: images-url
+      workspace: images-url
+
+  workspaces:
+  - name: images-url
+
+" | oc apply -f - > /dev/null
+
+oc describe pipeline simple-release
+
+
+title "Verify Push Secret"
+
+# Check if required secret exists
+oc get secret release-demo > /dev/null
```

Comment:
This just makes sure it fails if secret doesn't exist, right?

Reply:
Yup. I tried to express that in the comment. Should I try rephrasing it?